packages/participants-svc/src/domain/participant_agg.ts

```diff
@@ -3731,5 +3730,21 @@
 
          return existing;
      }
+
+    private async _getAccountBalance(paricipantId: string , account: IParticipantAccount):Promise<number> {
+        const accBalAccount = await this._accBal.getAccount(account.id);
+        if (!accBalAccount) {
+            throw new AccountNotFoundError(`Could not get participant ${paricipantId}'s ${account.type} account with id: ${account.id} from accounts and balances`);
+
+
+        }
+        return Number(accBalAccount.balance);
+    }
+
+    private async _validateRequestedNdcAmount(requestedNdcAmount: number, settlementAccountBalance: number, errorMessage: string) : Promise<void> {
+        if (settlementAccountBalance <= requestedNdcAmount) {
```

Comment:
How is "position" factored into this?

Is there a check somewhere to confirm that this new NDC value is not more than the current position?